CSCIE-599/HELM-Editor-UI
# End to End tests

Lots of end to end tests added, along with some minor functionality changes to aid in testing.

Tests added:
DONE:
load sequence tests
 - clicking load displays the modal dialog
 - clicking the x hides the modal dialog (except not in travis because firefox is annoying and broken)
 - clicking outside of the modal dialog hides the modal dialog
 - loading in an incorrect PEPTIDE sequence results in an error message
 - loading in an incorrect NUCLEOTIDE (RNA/DNA) sequence results in an error message
 - loading in a correct/known PEPTIDE sequence results in valid HELM and valid SVG display
 - loading in a correct/known NUCLEOTIDE sequence results in valid HELM and valid SVG display
 - loading in a known-incorrect HELM sequence results in an error message
 - loading in a known-correct HELM sequence results in the HELM string and valid SVG display
 - - test a single polymer
 - - test a HELM sequence with multiple polymers
 - - test a HELM sequence with a single cycle
 - - test a HELM sequence with two cycles - should result in warning

lower canvas pane tests
 - toggling the lower pane flips from HELM to Molecular Details to Sequence display
 - clicking on the image link brings up the modal dialog

monomer library scenarios
 - search for RNA
 - search for PEPTIDE
 - search for CHEM
 - explore for RNA
 - explore for PEPTIDE
 - explore for CHEM
 - click a monomer to select it, click to deselect
 - double click on monomer to get details
 - double click on a fragment to get the notation

editor scenarios
 - click-click add monomer to blank canvas
 - click-click add monomer to canvas with a sequence already there
 - delete node from middle of PEPTIDE sequence
 - delete node from beginning of PEPTIDE sequence
 - delete node from end of PEPTIDE sequence
 - delete node from CHEM sequence
 - delete ribose from beginning of an RNA sequence
 - delete phosphate from beginning of an RNA sequence
 - delete ribose from end of an RNA sequence
 - delete phosphate from end of an RNA sequence
 - delete ribose from middle of an RNA sequence
 - delete phosphate from middle of an RNA sequence
 - delete base node from an RNA sequence
